### PR TITLE
Python Question and Answers

### PR DESCRIPTION
**Pull request by AbdulKhadhar**
45 Python Question and answers

### CODE REVIEW

**Review by bregman-arie — CHANGES_REQUESTED** [review]
Some questions already exist in the repo + it's not adjusted to the format used right now

**Comment by AbdulKhadhar** [comment]
> 
> 
> Some questions already exist in the repo + it's not adjusted to the format used right now

please give a sample example of the format.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1 +1,271 @@
...
+    The allocation of Python heap space for Python objects is done by Python memory manager. The core API gives access to some tools for the programmer to code.
+    Python also have an inbuilt garbage collector, which recycle all the unused memory and frees the memory and makes it available to the heap space.
+
+6) What are the tools that help to find bugs or perform static analysis?
+
+PyChecker is a static analysis tool that detects the bugs in Python source code and warns about the style and complexity of the bug. Pylint is another tool that verifies whether the module meets the coding standard.
+
+7) What are Python decorators?
+
+A Python decorator is a specific change that we make in Python syntax to alter functions easily.
```
Comment (surister) [inline CONTRIBUTING.md]: That's rather vague, what is a decorator? what is its syntax? uses? ...

**Review by surister — COMMENTED** [review]
Not to mention that you committed  on the wrong file, correct file would be README.md
You can find the correct format in there.

Also would be awesome if you could add small snippets of code or make some more elaborate/complete answers especially to complex questions.

**Pull request closed without merging**